tests/sync/app.cpp

```diff
@@ -824,6 +827,350 @@ TEST_CASE("app: link_user integration", "[sync][app]") {
+                    CHECK(err.error_code.value() < 0);
+                }
+            }
+            CHECK(!error);
+        });
+        
+        realm::app::RemoteMongoCollection::RemoteFindOneAndModifyOptions find_and_modify_options {
+            util::Optional<std::string>(nlohmann::json({{ "name", "fido" }}).dump()), //project
+            util::Optional<std::string>(nlohmann::json({{ "name", 1 }}).dump()), //sort,
+            true, //upsert
+            true // return new doc
+        };
+        
+        collection.find_one_and_delete(dog_document, find_and_modify_options, [&](Optional<app::AppError> error) {
+            CHECK(!error);
+        });
+        
+        collection.find_one_and_delete(dog_document, [&](Optional<app::AppError> error) {
+            CHECK(!error);
+            processed = true;
+        });
+        
+        CHECK(processed);
+    }
+    
+    SECTION("count and aggregate") {
+        bool processed = false;
+        
+        std::vector<std::string> pipeline = {
+            "{ \"$match\": { \"name\": \"fido\" } }",
+            "{ \"$group\": { \"_id\": \"$name\" } }"
+        };
+        
+        collection.aggregate(pipeline, [&](std::string document_json, Optional<app::AppError> error) {
+            if (error) {
+                std::cout << "aggregate failed, message: " << error->error_code.message() << "+" << error->message << std::endl;
+            } else {
+                try {
+                    auto json = nlohmann::json::parse(document_json);
+                    CHECK(json.is_array());
+                } catch (AppError err) {
+                    CHECK(err.error_code.value() < 0);
+                }
+            }
+            CHECK(!error);
+        });
+        
+        collection.count("{\"breed\":\"husky\"}", [&](uint64_t count, Optional<app::AppError> error) {
+            CHECK(!error);
+            CHECK(count >= 1);
+        });
+        
+        collection.count("{\"breed\":\"husky\"}", 1, [&](uint64_t count, Optional<app::AppError> error) {
+            CHECK(!error);
+            CHECK(count == 1);
+            processed = true;
+        });
+        
+        CHECK(processed);
+    }
+    
+    SECTION("update") {
+        
+        bool processed = false;
+        
+        realm::app::RemoteMongoCollection::RemoteFindOneAndModifyOptions find_and_modify_options {
+            util::Optional<std::string>(nlohmann::json({{ "name", "fido" }}).dump()), //project
+            util::Optional<std::string>(nlohmann::json({{ "name", 1 }}).dump()), //sort,
+            true, //upsert
+            true // return new doc
+        };
+        
+        collection.find_one_and_update(dog_document, dog_document2, find_and_modify_options, [&](std::string document, Optional<app::AppError> error) {
+            if (error) {
+                std::cout << "find_one_and_update failed, message: " << error->error_code.message() << "+" << error->message << std::endl;
```

Comment:
These need to produce test failures rather than just printing a message.

Reply:
removed message